scripts/obs-status/obs-status.rb

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/ruby
+
+require "optparse"
+require "open3"
+require "active_support/core_ext/hash/conversions"
+
+class ObsHandler
+  def initialize(config = {})
+    @group = config[:group] || "cloud-maintenance"
+  end
+
+  def requests()
+    result = obs_api_call("/search/request?match=review[@by_group='#{@group}'+and+(@state='new')]")
+    requests = result["request"]
+    open_requests = []
+    requests.each do |request|
+      if request["state"]["name"] = "review" && request["id"]
+        id = request["id"].to_i
+        status = request_status?(request["id"])
+        source = request["action"][1]["source"]["project"].gsub!(/:/, "/")
+        open_requests.push(
+          {
+            id: id,
+            source: source,
+            status: status
+          }
+        )
+      end
+    end
+    open_requests
+  end
+
+  def request_status?(id)
```

Comment:
If we use the new obs staging workflow (or the osc staging plugin) this information will be encoded in if the submit request is assigned to a staging project.
running <- Assigned to the gating staging project. And gate run is in progress.
failure <- Assigned to the gating staging project. And gate run is not in progress.
success <- Request was accepted.
pending <- Open and not assigned to the gating staging project.
Whether a request already had a run might be encoded in a comment. The previous obs staging support added this and if a run is in progress as structured information to obs.

Are you suggesting we should not use one of these staging workflows but build our own?

Reply:
Scratch that comment, it didn't occur to me what _release_ request meant. I.e. the submit request that releases the changes in a SUSE:Maintenance:ID project to the product updates project. There the maintenance project already does the job of a staging project.

Instead:

Please make it clear in the --help output what this is intended for.